### ISSUE TITLE
In silico USI Proposal

### ISSUE THREAD

**Issue by mwang87** [mwang87/MetabolomicsSpectrumResolver#173]
So there is some need to think about in silico MS/MS from a variety of proteomics and metabolomics sources. With real data, there is no need to think about parameters in generation because the data is data, however with in silico and the differing types of parameters depending upon the tool, requires us to think about how do we encode these succinctly but also expressively. 

## Metabolomics

CFM-ID 4.0

## Proteomics

Prosit
MS2PIP

**Comment by sdrogers** [comment]
I’m a little wary of them being included at all tbh - quality is really not fully demonstrated (for metab at least). Although one benefit of including would make those comparisons easier.

that said, is there really a distinction between these generation params and MS acquisition params? For the latter, all those params are captured in the database entry that the USI resolves to. Wouldn’t the same count for the former? Unless you were proposing generating them on the fly? But if so, the spectrum is no longer permanent (as the generating algorithms change)?

**Comment by mwang87** [comment]
So my thought was that it would be generated on the fly as you wouldn't know which compounds to generate a priori. Broadly, my own interest is exactly as you mentioned, how do we do comparisons. 

I was proposing to generate them on the fly and the lack of a data backing requirement, does introduce the issue of permanence. 

I think an alternative approach for what I personally want to achieve (god I love the niceness of the visualizations) is the ability to have people enter their own peaks to plot via API.

**Comment by sdrogers** [comment]
Are the in silico tools properly versioned or dockerised? A version (or image tag) could be used in the usi?

If not, could dockerise?

**Comment by mwang87** [comment]
Yes, many are dockerized, there is an issue of namespacing both the version of the software and the model. I think that is what the library name area would be useful for.
